tools/mpremote/mpremote/main.py

```diff
@@ -365,6 +407,26 @@ def execbuffer(pyb, buf, follow):
     return ret_val
 
 
+def print_help():
+    def print_command_help(cmd, help_message, max_command_len):
+        help_message_lines = dedent(help_message).split("\n")
+        help_message = help_message_lines[0]
+        for line in help_message_lines[1:]:
+            help_message = "{}\n{}{}".format(help_message, " " * (max_command_len + 4), line)
+        print("  ", cmd, " " * (max_command_len - len(cmd) + 2), help_message, sep="")
+
+    print(_PROG, "-- MicroPython remote control")
+    print("\nList of commands:")
+    max_command_length = max(len(cmd) for cmd in _COMMANDS.keys())
+    for cmd in sorted(_COMMANDS.keys()):
+        print_command_help(cmd, _COMMANDS[cmd][3], max_command_length)
+
+    print("\nList of shortcuts:")
+    max_command_length = max(len(cmd) for cmd in _command_expansions.keys())
+    for cmd in sorted(_command_expansions.keys()):
+        print_command_help(cmd, _command_expansions[cmd][2], max_command_length)
```

Comment:
this and the above section of code share a lot of code, which could be factored into the `print_command_help` function; ie pass in the dict (either `_COMMANDS` or `_command_expansions`) and the index of the help string (3 or 2)

Reply:
Thank you, fixed